opamp/observiq/reload_funcs.go

```diff
@@ -0,0 +1,262 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observiq
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/observiq/observiq-otel-collector/internal/logging"
+	"github.com/observiq/observiq-otel-collector/opamp"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+func managerReload(client *Client, managerConfigPath string) opamp.ReloadFunc {
+	return func(contents []byte) (bool, error) {
+		// Unmarshal config and only pull fields out that are allowed to be updated.
+		var newConfig opamp.Config
+		if err := yaml.Unmarshal(contents, &newConfig); err != nil {
+			return false, fmt.Errorf("failed to validate config %s", ManagerConfigName)
+		}
+
+		// Check if the updatable fields are equal
+		// If so then exit
+		if client.currentConfig.CmpUpdatableFields(newConfig) {
```

Comment:
also, because of this, if we have different hashes but the correct config we will hit this over and over again never correctly. in this case would it be better to only overwrite the file but not mess with hot reloading?

Reply:
Every time we have have an incorrect hash we will reload, giving it another chance to correctly compute. If it's unlikley that we are able to fully do a reload then fail the hash since there are so many file ops present in the reload.

Reply:
talked about this offline. not a major concern and no need to change anything